cql3/restrictions/statement_restrictions.cc

```diff
@@ -372,6 +372,7 @@ statement_restrictions::statement_restrictions(database& db,
     const expr::allow_local_index allow_local(
             !_partition_key_restrictions->has_unrestricted_components(*_schema)
             && _partition_key_restrictions->is_all_eq());
+    _has_multi_column = find_atom(_clustering_columns_restrictions->expression, expr::is_multi_column);
```

Comment:
I can't comment on the commit message so I'll do it here:
Reading the commit message, it seems like this is some sort of optimization of whether or not to use the index. But the linked issue suggests this is actually a correctness bug! So the commit message should have mentioned the error, not just the optimization angle.